octoprint_mrbeam/iobeam/laserhead_handler.py

```diff
@@ -325,8 +326,13 @@ def _load_laser_heads_file(self):
                     if "last_used_lh_serial" in data:
                         self._last_used_lh_serial = data["last_used_lh_serial"]
 
-                    if "last_used_lh_model" in data:
-                        self._last_used_lh_model = data["last_used_lh_model"]
+                    if "last_used_lh_model_string" in data:
+                        self._last_used_lh_model_string = data[
+                            "last_used_lh_model_string"
+                        ]
+
+                    if "last_used_lh_model_id" in data:
+                        self._last_used_lh_model_id = data["last_used_lh_model_id"]
 
                     if "correction_settings" in data:
                         self._correction_settings = data["correction_settings"]
```

Comment:
Please refactor to remove all of these if statements

Reply:
And remove the general try / except as well. You may catch OSError or similar which can happen when the file is not present. You may also catch errors when the Yaml file has incorrect syntax.
And to prevent that error from happening, make sure to use `os.path.isfile`.

Reply:
So why I should remove it then? If errors can appera here we should catch them

Reply:
yes, but only the errors you foresee to happen. catching all exceptions are an absolute no-no otherwise.

Reply:
you can use `@logExceptions` to log and re-raise the exceptions

Reply:
As you can see the exceptions are caught everywhere...
```python
    def _handle_laserhead(self, dataset):
        """
        Handle laserhead dataset, iobeam sends the whole laserhead data.
        :param dataset:
        :return: error count
        """
        try:
            self._laserhead_handler.set_current_used_lh_data(dataset)
        except:
            self._logger.exception("laserhead: exception while handling head:data: ")
        return 0
```

Reply:
you can also have a look at https://stackoverflow.com/a/26199212/11136955

Reply:
> And remove the general try / except as well. You may catch OSError or similar which can happen when the file is not present. You may also catch errors when the Yaml file has incorrect syntax. And to prevent that error from happening, make sure to use `os.path.isfile`.

This was already done